plugins/csv_dataset.cc

```diff
@@ -575,7 +575,13 @@ struct CsvDataset::Itl: public TabularDataStore {
         filename = config.dataFileUrl.toString();
         
         // Ask for a memory mappable stream if possible
-        ML::filter_istream stream(filename, { { "mapped", "true" } });
+        ML::filter_istream stream;
+        try {
+            stream.open(filename, { { "mapped", "true" } });
+        }
+        catch (const std::exception & e) {
+             throw HttpReturnException(400, "dataFileUrl pointed to an empty file");
```

Comment:
Finally, the filter_istream should not fail to map an empty file; it should return a zero length mapping.  Once that is fixed, the check for zero length can be made elsewhere (and @nicolaskruchten does it make sense to disallow an empty csv file?  One with no rows and no columns is empty, but a valid (if useless) CSV file.